lib/ansible/modules/cloud/azure/azure_rm_trafficmanagerprofile.py

```diff
@@ -0,0 +1,451 @@
+
+                if update_tags:
+                    to_be_updated = True
+
+                to_be_updated = to_be_updated or self.check_update(response)
+
+            if to_be_updated:
+                self.log("Need to Create / Update the Traffic Manager profile")
+
+                if not self.check_mode:
+                    self.results = shorten_traffic_manager_dict(self.ceate_update_traffic_manager_profile())
+                    self.log("Creation / Update done.")
+
+                self.results['changed'] = True
+                return self.results
+
+        elif self.state == 'absent' and response:
+            self.log("Need to delete the Traffic Manager profile")
+            self.results['changed'] = True
+
+            if self.check_mode:
+                return self.results
+
+            self.delete_traffic_manager_profile()
+
+            self.log("Traffic Manager profile deleted")
+
+        return self.results
+
+    def get_traffic_manager_profile(self):
+        '''
+        Gets the properties of the specified Traffic Manager profile
+
+        :return: deserialized Traffic Manager profile dict
+        '''
+        self.log("Checking if Traffic Manager profile {0} is present".format(self.name))
+        try:
+            response = self.traffic_manager_management_client.profiles.get(self.resource_group, self.name)
+            self.log("Response : {0}".format(response))
+            self.log("Traffic Manager profile : {0} found".format(response.name))
+            self.endpoints_copy = response.endpoints if response and response.endpoints else None
+            return traffic_manager_profile_to_dict(response)
+        except CloudError:
+            self.log('Did not find the Traffic Manager profile.')
+            return False
+
+    def delete_traffic_manager_profile(self):
+        '''
+        Deletes the specified Traffic Manager profile in the specified subscription and resource group.
+        :return: True
+        '''
+
+        self.log("Deleting the Traffic Manager profile {0}".format(self.name))
+        try:
+            operation_result = self.traffic_manager_management_client.profiles.delete(self.resource_group, self.name)
+            return True
+        except CloudError as e:
+            self.log('Error attempting to delete the Traffic Manager profile.')
+            self.fail("Error deleting the Traffic Manager profile: {0}".format(e.message))
+            return False
+
+    def ceate_update_traffic_manager_profile(self):
+        '''
+        Creates or updates a Traffic Manager profile.
+
+        :return: deserialized Traffic Manager profile state dictionary
+        '''
+        self.log("Creating / Updating the Traffic Manager profile {0}".format(self.name))
+
+        parameters = Profile(
+            tags=self.tags,
+            location=self.location,
+            profile_status=self.profile_status,
+            traffic_routing_method=self.routing_method,
```

Comment:
Will this still work with the lowercase first char that was changed?

Reply:
yes

Reply:
covered by test